api/rql/internal/predicate/boolean.go

```diff
@@ -16,11 +19,21 @@ type boolean struct {
 }
 
 func (p *boolean) Marshal() interface{} {
-	return p.val
+	return []interface{}{"boolean", p.val}
 }
 
 func (p *boolean) Unmarshal(input interface{}) error {
-	val, ok := input.(bool)
+	if !matcher.Array(matcher.Value("boolean"))(input) {
+		return errz.MatchErrorf("must be formatted as ['boolean', <value>]")
```

Comment:
I'm not really sure why the type name is needed, as JSON has boolean types.

Reply:
Keep symmetric w/ the other types (only if said type has more than one possible value). We have "string", "number", "object", "array". "boolean" has two values true/false, so it made sense to me to phrase it that way. `nil` has only one value, so that's why it's labeled as such.

I guess I can go back to `true`/`false` if that's preferred too.

Reply:
I think I'd like to see all of those as a set of examples before deciding. All of those have types in JSON, so I don't remember why we need to call out their types.